src/Fixer/LanguageConstruct/SingleSpaceAroundConstructFixer.php

```diff
@@ -0,0 +1,462 @@
+                if ($tokens[$whitespaceTokenIndex]->equals([T_WHITESPACE]) && str_contains($tokens[$whitespaceTokenIndex]->getContent(), "\n")) {
+                    continue;
+                }
+            }
+
+            $tokens->ensureWhitespaceAtIndex($whitespaceTokenIndex, 0, ' ');
+        }
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        $tokenMapContainASingleSpaceKeys = array_keys(self::$tokenMapContainASingleSpace);
+        $tokenMapPrecededByASingleSpaceKeys = array_keys(self::$tokenMapPrecededByASingleSpace);
+        $tokenMapFollowedByASingleSpaceKeys = array_keys(self::$tokenMapFollowedByASingleSpace);
+
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('constructs_contain_a_single_space', 'List of constructs which must contain a single space.'))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([new AllowedValueSubset($tokenMapContainASingleSpaceKeys)])
+                ->setDefault($tokenMapContainASingleSpaceKeys)
+                ->getOption(),
+            (new FixerOptionBuilder('constructs_preceded_by_a_single_space', 'List of constructs which must be preceded by a single space.'))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([new AllowedValueSubset($tokenMapPrecededByASingleSpaceKeys)])
+                ->setDefault($tokenMapPrecededByASingleSpaceKeys)
+                ->getOption(),
+            (new FixerOptionBuilder('constructs_followed_by_a_single_space', 'List of constructs which must be followed by a single space.'))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([new AllowedValueSubset($tokenMapFollowedByASingleSpaceKeys)])
+                ->setDefault($tokenMapFollowedByASingleSpaceKeys)
+                ->getOption(),
+        ]);
+    }
+
+    private function isMultiLineReturn(Tokens $tokens, int $index): bool
+    {
+        ++$index;
+        $tokenFollowingReturn = $tokens[$index];
+
+        if (
+            !$tokenFollowingReturn->isGivenKind(T_WHITESPACE)
+            || !str_contains($tokenFollowingReturn->getContent(), "\n")
+        ) {
+            return false;
+        }
+
+        $nestedCount = 0;
+
+        for ($indexEnd = \count($tokens) - 1, ++$index; $index < $indexEnd; ++$index) {
+            if (str_contains($tokens[$index]->getContent(), "\n")) {
+                return true;
+            }
+
+            if ($tokens[$index]->equals('{')) {
+                ++$nestedCount;
+            } elseif ($tokens[$index]->equals('}')) {
+                --$nestedCount;
+            } elseif (0 === $nestedCount && $tokens[$index]->equalsAny([';', [T_CLOSE_TAG]])) {
+                break;
+            }
+        }
+
+        return false;
+    }
+
+    private function isMultilineExtendsOrImplementsWithMoreThanOneAncestor(Tokens $tokens, int $index): bool
```

Comment:
This looks like something that may be helpful in other fixers, so maybe we can extract it somewhere? Same with `isMultilineConstant()`.

Reply:
same as https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6857#discussion_r1148442540